### PR TITLE
Build: Remove hardcoded reference to x-plugins in build

### PR DESCRIPTION
**Pull request by rjernst**
The gradle build currenlty allows extra plugins to be hooked into the
elasticsearch build by placing under an x-plugins directory as a sibling
of the elasticsearch checkout. This change converts this directory to
one called elasticsearch-extra. The subdirectories of
elasticsearch-extra become first level projects in gradle (while before
they would have been subprojects of `:x-plugins`. Additionally, this
allows major version checkouts to be associated with extra plugins. For
example, you could have elasticsearch checked out as
`elasticsearch-5.x`, and a sibling `elasticsearch-5.x-extra` directory.

### CODE REVIEW

**Comment by clintongormley** [comment]
So would you clone x-plugins as `elasticsearch-extra`, or would you clone x-plugins inside the `elasticsearch-extra/` dir?

**Comment by rjernst** [comment]
The latter: x-plugins (or any other extra gradle project) could be subdirectories `elasticsearch-extra` and be automatically added as subprojects in the elasticsearch build.

File: settings.gradle [hunk 1]
```diff
@@ -96,17 +96,19 @@ void addSubProjects(String path, File dir) {
...
-if (xplugins.exists()) {
-  include ':x-plugins'
-  project(':x-plugins').projectDir = xplugins
-  for (File extraPluginDir : xplugins.listFiles()) {
-    addSubProjects(':x-plugins', extraPluginDir)
+String dirName = rootProject.projectDir.name
+File extraProjects = new File(rootProject.projectDir.parentFile, "${dirName}-extra")
+if (extraProjects.exists()) {
+  for (File extraProjectDir : extraProjects.listFiles()) {
+    addSubProjects('', extraProjectDir)
```
Comment (pickypg) [inline settings.gradle]: Does something else use `path`? Since we're blanking it out, it seems like we can remove that parameter now.

Comment (pickypg) [inline settings.gradle]: Ah, it's recursive. Nevermind.

**Review by pickypg — APPROVED** [review]
LGTM

**Comment by clintongormley** [comment]
@rjernst should we get this change in now?

**Comment by ywelsch** [comment]
Can you update the docs as well? TESTING.asciidoc still refers to x-plugins directory.

**Comment by rjernst** [comment]
@ywelsch I updated the testing doc in 360dc4d.

**Pull request merged**